mio/process/video.py

```diff
@@ -156,13 +156,27 @@ def process_frame(self, input_frame: np.ndarray) -> Optional[np.ndarray]:
         if input_frame is None:
             return None
 
-        if self.noise_patch_config.enable and self.previous_frame is not None:
-            # Call the unified detection method
-            broken, noise_patch = self.noise_detect_helper.detect_frame_with_noisy_buffer(
-                input_frame,
-                self.previous_frame if self.noise_patch_config.method == "mean_error" else None,
-                self.noise_patch_config,
-            )
+        if self.noise_patch_config.enable:
+            if self.noise_patch_config.method == "gradient":
+                # For the gradient method, analyze the first frame immediately
+                broken, noise_patch = self.noise_detect_helper.detect_frame_with_noisy_buffer(
+                    input_frame,
+                    None,  # No previous frame needed for gradient method
+                    self.noise_patch_config,
+                )
+            else:
+                # For the mean method, wait until the second frame to start analysis
+                if self.previous_frame is not None:
+                    broken, noise_patch = self.noise_detect_helper.detect_frame_with_noisy_buffer(
+                        input_frame,
+                        self.previous_frame,
+                        self.noise_patch_config,
+                    )
+                else:
+                    # If it's the first frame and the method is mean_error, just store the frame
+                    self.previous_frame = input_frame
+                    self.append_output_frame(input_frame)
+                    return input_frame
 
```

Comment:
Noticed it got outdated, so just updating here:
```suggestion
            else:
                raise NotImplementedError(f"Noise detection method {self.noise_patch_config.method} is not implemented.")
```